sceptre/diffing/stack_differ.py

```diff
@@ -159,13 +208,37 @@ def _create_deployed_stack_config(self, stack_actions: StackActions) -> Optional
                 role_arn=stack.get('RoleARN')
             )
 
-    def _remove_default_parameters_that_arent_passed(
+    def _handle_special_parameter_situations(
         self,
         stack_actions: StackActions,
         generated_config: StackConfiguration,
         deployed_config: StackConfiguration
     ):
-        deployed_config_default_map = self._get_parameter_default_map(stack_actions)
+        deployed_template_summary = stack_actions.fetch_remote_template_summary()
+        generated_template_summary = stack_actions.fetch_local_template_summary()
+
+        if deployed_config is not None:
+            # If the parameter is not passed by Sceptre and the value on the deployed parameter is
+            # the default value, we'll actually remove it from the deployed parameters list so it
+            # doesn't show up as a false positive.
+            self._remove_deployed_default_parameters_that_arent_passed(
+                deployed_template_summary,
+                generated_config,
+                deployed_config
+            )
+        if not self.show_no_echo:
+            # We don't actually want to show parameters Sceptre is passing that the local template
+            # marks as NoEcho parameters (unless show_no_echo is set to true). Therefore those
+            # parameter values will be masked.
+            self._mask_no_echo_parameters(generated_template_summary, generated_config)
```

Comment:
@zaro0508 This handles the situation we were recently discussing with NoEcho parameters.